include/boost/sam/detail/semaphore_impl.hpp

```diff
@@ -28,7 +28,7 @@ struct semaphore_impl : detail::service_member
   }
 
   semaphore_impl &operator=(const semaphore_impl &) = delete;
-  semaphore_impl &operator=(semaphore_impl &&lhs) noexcept
+  semaphore_impl &operator=(semaphore_impl &&lhs)
```

Comment:
ugh, non-noexcept move - do we not have another option for this?

Reply:
no, mutex needs to lock.